sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -47,6 +37,7 @@ impl Default for StagedProgressBar {
 }
 
 impl StagedProgressBar {
```

Comment:
Should we abstract our progressbar needs into a trait and do a bit more of a general refactoring?  There are a mix of progressbar users throughout the codebase taking either `impl Fn()` or `&dyn Fn() + 'static`.  These would probably all be better off as a `&dyn Trait` that had functions to signal level of progress or change stages.

We could also provide a `NoProgressBar` impl of the trait which did nothing rather than the current practice of passing in the lambda `|_, _| {}`.

WDYT?